src/node.ts

```diff
@@ -610,6 +626,22 @@ export class Node implements IBehaviorAware<Node> {
         return results;
     }
 
+    /**
+     * Get all child-meshes of this node
+     * @param directDescendantsOnly defines if true only direct descendants of 'this' will be considered, if false direct and also indirect (children of children, an so on in a recursive manner) descendants of 'this' will be considered (Default: false)
+     * @param predicate defines an optional predicate that will be called on every evaluated child, the predicate must return true for a given child to be part of the result, otherwise it will be ignored
+     * @returns an array of AbstractMesh
+     */
+    public getChildMeshes<T extends AbstractMesh>(directDescendantsOnly?: boolean, predicate?: (node: Node) => node is T): T[];

```

Comment:
I am really wondering if I could not use this in the wrong way let s say I call it with: 

```
getChildMeshes<Mesh>(false, (am) => true);
```

TS won t be able to detect I am not using a typeguard here and instancedMeshes would get mixed in return as Mesh type. I am really wondering if the overload is safe enough ?

Reply:
Should actually be totally fine in the latest TS version which is awesome :-) 

https://www.typescriptlang.org/play?#code/MYGwhgzhAEBCkFMByB7AJg6BvAUNf0ADgK4BGIAlsNAE4JhooB2IAntMABYUhp1MAuOIlQYA2gF1oAXmiSA3DjwES5KtADmCAC4BhbrwCyCCJxMAeACrQEAD20ImaGAEFSEbTTDBtx0wD4ACkI6NCowByFApnQEIXgIZFiAShl-aBiMaAoYS2ShSwUlAiIySmotPQM0PzMIYNDwyOho2PiRFLToUhQUEHomfOg3Dy8fWqLlfFVyzR19HhqTOoaEMOAIuJbMrYSkjFTpdJ6+gaGRz29fZclsKZKOZg9aE2IQbQghFxovVnMLsbXAIyOQSRQPAgAMxQNBawCe2g41WgKEh0G03AgADouIt+KlcBCIRQ0YFArjeNBIMN3JdxstkliAPpgdwAZUICGAFDAIAAYhQEJSAGTCoiNDYOcnVZIE+5Eh50CBvD5YkimaWLKmuWmA2rJcEKggAX3lJrN+Do2mINCYL2V7wghvwptNOFAkB1oyutRs9kczmEiVEmEJKjK6joDGYbGgLPZnO5vIFQrQIM8xAQijdHqg0F9dgcTi9dKBnDuJRmkfojBY7CZAFtlhyuTz+YLKbIM1mcDnwHmAJJMDxgJjANYF-3Fmne+mmCvhtTUKO12NMihN0wtpPt1PpmiZ7NKeHDxENkFMBAAd3zy0CBvdCOy59kl5vQ5HY4nd4fj9P3TtV9ryDfYEHvRRSCYHFqn4NViA1BsH0g6C8UcOCNQ3X8T2eJUVRBZDKgWIxlgsWogkCJlDnSbtfyAA

Reply:
TS is able to detect you are not using a type guard and gives you an error:
`Signature '(node: Node): boolean' must be a type predicate.(2345)`
This is exactly the same pattern that is used for `Array.filter`.